lib/galaxy/jobs/runners/univa.py

```diff
@@ -157,11 +156,10 @@ def _get_drmaa_state_qstat(self, job_id, extinfo):
         # even if this could be disambiguated by the stderr message the `qstat -u "*"`
         # way seems more generic
         cmd = ['qstat', '-u', '"*"']
-        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        stdout, stderr = p.communicate()
-        stdout, stderr = unicodify(stdout).strip(), unicodify(stderr).strip()
-        if p.returncode != 0 or stderr != "":
-            log.exception('`%s` returned %d, stderr: %s' % (' '.join(cmd), p.returncode, stderr))
+        try:
+            stdout = commands.execute(cmd)
+        except commands.CommandLineException as e:
+            log.exception(str(e))
```

Comment:
That's going to log the traceback and the nicely formatted `CommandLineException `, which is redundant. Maybe use `log.error` ?

Reply:
Actually this is redundant either way:
```python
In [15]: try:
    ...:     raise commands.CommandLineException('echo', 'bla', 'blurb', 1)
    ...: except Exception as e:
    ...:     log.exception(e)
    ...:
    ...:
Failed to execute command-line echo, stderr was:
-------->>begin stderr<<--------
blurb
-------->>end stderr<<--------
-------->>begin stdout<<--------
bla
-------->>end stdout<<--------
Traceback (most recent call last):
  File "<ipython-input-15-8f4a354ec6e3>", line 2, in <module>
    raise commands.CommandLineException('echo', 'bla', 'blurb', 1)
galaxy.tool_util.deps.commands.CommandLineException: Failed to execute command-line echo, stderr was:
-------->>begin stderr<<--------
blurb
-------->>end stderr<<--------
-------->>begin stdout<<--------
bla
-------->>end stdout<<--------
```

Given the raise in the line below I'd maybe go with log.error(str(e)),
which would given you something like
```python
In [16]: try:
    ...:     raise commands.CommandLineException('echo', 'bla', 'blurb', 1)
    ...: except Exception as e:
    ...:     log.error(str(e))
    ...:
    ...:
Failed to execute command-line echo, stderr was:
-------->>begin stderr<<--------
blurb
-------->>end stderr<<--------
-------->>begin stdout<<--------
bla
-------->>end stdout<<--------
```

Reply:
done. did the same for drmaa.py